dimod/include/dimod/adjmapbqm.h

```diff
@@ -95,108 +97,110 @@ class AdjMapBQM {
     template <class B2>
     AdjMapBQM(B2 dense[], size_type num_variables,
               bool ignore_diagonal = false) {
-      // we know how big our linear is going to be
-      adj.resize(num_variables);
+        // we know how big our linear is going to be
+        adj.resize(num_variables);
 
-      // Backup copy of the diagonal of the dense matrix.
-      std::vector<B2> dense_diagonal(num_variables);
+        // Backup copy of the diagonal of the dense matrix.
+        std::vector<B2> dense_diagonal(num_variables);
 
-      if (!ignore_diagonal) {
-        #pragma omp parallel for
-        for (size_type v = 0; v < num_variables; ++v) {
-          adj[v].second = dense[v * (num_variables + 1)];
-        }
-      }
-
-      #pragma omp parallel
-      {
-        // Zero out the diagonal to avoid expensive checks inside innermost
-        // loop in the code for reading the matrix. The diagonal will be
-        // restored so a backup copy is saved.
-        #pragma omp for schedule(static)
-        for (size_type v = 0; v < num_variables; ++v) {
-          dense_diagonal[v] = dense[v * (num_variables + 1)];
-          dense[v * (num_variables + 1)] = 0;
+        if (!ignore_diagonal) {
+#pragma omp parallel for
```

Comment:
Annoys me that it cannot handle `#pragma omp parallel for` etc. There [are solutions](https://stackoverflow.com/questions/31353748/how-could-i-indent-c-pragma-using-clang-format) but probably not common enough to require.